capella2polarion/converters/converter_config.py

```diff
@@ -52,7 +52,50 @@ class CapellaTypeConfig:
 
     def __post_init__(self):
         """Post processing for the initialization."""
-        self.converters = _force_dict(self.converters)
+        self.converters = self._force_dict()
+
+    def _force_dict(self) -> dict[str, dict[str, t.Any]]:
+        match self.converters:
+            case None:
+                return {}
+            case str():
+                return {self.converters: {}}
+            case list():
+                return {c: {} for c in self.converters}
+            case dict():
+                return self._filter_converter_config()
+            case _:
+                raise TypeError("Unsupported Type")
+
+    def _filter_converter_config(self) -> dict[str, dict[str, t.Any]]:
+        custom_converters = (
+            "include_pre_and_post_condition",
+            "linked_text_as_description",
+            "add_attributes",
+            "add_context_diagram",
+            "add_tree_diagram",
+            "add_jinja_fields",
+            "jinja_as_description",
+        )
+        filtered_config = {}
+        assert isinstance(self.converters, dict)
+        for name, params in self.converters.items():
+            params = params or {}
+            if name not in custom_converters:
+                logger.error("Unknown converter in config %r", name)
+                continue
+
+            if name in ("add_context_diagram", "add_tree_diagram"):
+                assert isinstance(params, dict)
+                params = _filter_context_diagram_config(params)
+
+            if name in ("add_attributes"):
```

Comment:
Note that this is a string, not a 1-tuple, so this condition performs a substring check (as opposed to the 2-tuple of strings above). For example, the string "at" would match this condition, which is probably not what was intended.